src/main/java/uk/gov/ons/ctp/integration/ratelimiter/client/RateLimiterClient.java

```diff
@@ -307,6 +253,88 @@ private LimitDescriptor createLimitDescriptor(
     return limitDescriptor;
   }
 
+  /**
+   * Call the rate limiter using a circuit breaker. This will return without exception if 1) the
+   * request is within the rate limits, or 2) the call to the rate limiter fails in some way, or 3)
+   * due to previous failures the circuit breaker is 'open'. If the request is above the rate limits
+   * then a ResponseStatusException is thrown.
+   */
+  private void invokeRateLimiter(String requestDescription, RateLimitRequest request) {
+    ResponseStatusException limitException =
+        circuitBreaker.run(
+            () -> {
+              try {
+                doInvokeRateLimiter(requestDescription, request);
+                return null;
+              } catch (CTPException e) {
+                // we should get here if the rate-limiter is failing or not communicating
+                // ... wrap and rethrow to be handled by the circuit-breaker
+                throw new RuntimeException(e);
+              } catch (ResponseStatusException e) {
+                // we have got a 429 but don't rethrow it otherwise this will count against
+                // the circuit-breaker accounting, so instead we return it to later throw
+                // outside the circuit-breaker mechanism.
+                return e;
+              }
+            },
+            throwable -> {
+              // This is the Function for the circuitBreaker.run second parameter, which is called
+              // when an exception is thrown from the first Supplier parameter (above), including
+              // as part of the processing of being in the circuit-breaker OPEN state.
+              //
+              // It is OK to carry on, since it is better to tolerate limiter error than fail
+              // operation, however by getting here, the circuit-breaker has counted the failure,
+              // or we are in circuit-breaker OPEN state.
+              if (throwable instanceof CallNotPermittedException) {
+                log.info("Circuit breaker is OPEN calling rate limiter for " + requestDescription);
```

Comment:
no code coverage for this. Please add a couple of unit tests that throws CallNotPermittedException (this was in the original RHSVc unit tests).  Ideally a unit test in each of fulfilment and webform paths.

Reply:
Tests added for both variants.